srv/components/misc_pkgs/kafka/start.sls

```diff
@@ -26,23 +26,39 @@ Start zoopkeper:
     - name: kafka-zookeeper
     - enable: True
 
-#TODO: find better solution to add delay
-Wait for zookeeper to start:
-  module.run:
-    - test.sleep:
-      - length: 10
+Ensure kafka-zookeeper has started:
+  cmd.run:
+    - name: test 1 -le $(ps ax | grep java | grep -i QuorumPeerMain | grep -v grep | awk '{print $1}' | wc -l)
+    - retry:
+    # Ref: https://docs.saltproject.io/en/3000/ref/states/requisites.html#retrying-states
+        attempts: 10
+        until: True
+        interval: 2
     - require:
       - Start zoopkeper
 
-#Start kafka:
+# Start kafka:
 #  cmd.run:
 #    - name: ./bin/kafka-server-start.sh -daemon config/server.properties
 #    - cwd: /opt/kafka
 #    - unless: test 1 -le $(ps ax | grep ' kafka\.Kafka ' | grep java | grep -v grep | awk '{print $1}' | wc -l)
+#    - require:
+#       - Ensure kafka-zookeeper has started
 
 Start kafka:
   service.running:
     - name: kafka
     - enable: True
     - require:
-      - Wait for zookeeper to start
+      - Ensure kafka-zookeeper has started
+
+Ensure kafka has started:
+  cmd.run:
+    - name: test 1 -le $(ps ax | grep -i 'kafka.Kafka' | grep -v grep | awk '{print $1}' | wc -l)
```

Comment:
Use service command

Reply:
@sachinpunadikar the section on line 53 takes care of this requirement.
```
Start kafka:
  service.running:
    - name: kafka
    - enable: True
    - require:
      - Ensure kafka-zookeeper has started
 ```
Salt ensures that service has actually started in this block.

We are checking if the process has started as additional assurance. Plus, there is another check against endpoint here: https://github.com/Seagate/cortx-prvsnr/pull/1215/files#diff-37b79a5616fb629e878b55f74b7d7dea6b4bc722c54d74b57a6e5bd25964ae6fR20